src/AutoMapper/Internal/IMappingOptionsMemberMapper.cs

```
@@ -289,7 +289,7 @@ public class NameSplitMember : IChildMemberConfiguration
         public INamingConvention SourceMemberNamingConvention { get; set; }
         public INamingConvention DestinationMemberNamingConvention { get; set; }
 
-        public IEnumerable<MethodInfo> SourceExtensionMethods => (Mapper.Configuration as ConfigurationStore).SourceExtensionMethods;
+        public IEnumerable<MethodInfo> SourceExtensionMethods => (Mapper.Configuration as MapperConfiguration).SourceExtensionMethods;
```

I think a cast would work better here. Let it crash if it's the wrong type.